### PR TITLE
Add docker support for development environment

### LINKED ISSUE #1074: Add docker support
Getting started with epson on a vanilla Mac is not trivial due to third-party dependencies (e.g., cairo). Having a Dockerfile and/or a docker compose would be helpful in these cases.

### PR DESCRIPTION
**Pull request by aramboi**
Other changes:

- Consolidate the requirements files and removed some obsolete dependencies.
- Consolidate test setup into `setup.cfg`.
- Remove obosolete files related to vagrant.
- Update Makefile with docker powered commands.
- Update README.
- Should fix dependabot issues like #1058 

NOTE: This is a pretty opinionated setup for a docker development environment, so reviews are welcome.

Closes #1074
Related PRs: #772 #1075 (should be closed after this is merged)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
-before_install:
-  - sudo apt-get install graphviz
-  
+# before_install:
+#   - sudo apt-get install graphviz
```
Comment (aramboi) [inline .travis.yml]: Why was this installed in the pipeline? Tests seem to run fine without it.

File: .flake8 [hunk 2]
```diff
@@ -1,8 +0,0 @@
-[flake8]
-# E272 - spaces before keyword
-# E221 – spaces before operator
-# E222 – spaces after operator
-# E251 – spaces after operator
-# E241 - multiple spaces after ':'
-# E131 - continuation line unaligned for hanging indent
-ignore = E221, E272, E241, W503, E251, E222, E131
```
Comment (aramboi) [inline .flake8:8]: This was moved in `setup.cfg`.

File: pycon/settings.py [hunk 3]
```diff
@@ -151,7 +150,7 @@ def _(x):
 )
 
 # Make this unique, and don't share it with anybody.
-SECRET_KEY = config('SECRET_KEY', default='your-secret-key')
+SECRET_KEY = config('SECRET_KEY')
```
Comment (aramboi) [inline pycon/settings.py:153]: This should make it harder to deploy in any environment with an insecure secret key.

**Comment by umgelurgel** [comment]
Thanks for your work on cleaning this up! Looking through it, the changes look great - I'll want to run through it locally to make sure everything's working as expected and then should be ready to merge. @artcz do you have any comments?

Note: update the deployment script to no longer use the `make install`.

**Review by umgelurgel — APPROVED** [review]
Thanks for making this work! Looks great.

**Pull request merged**